File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = {};
        answer = new int[commands.length];
        for (int i=0; i<commands.length; i++) {
            int a = commands[i][0];
            int b = commands[i][1];
            int k = commands[i][2];
            List<Integer> list = new ArrayList<>();
            for (int j=a-1;j<b; j++) {
                list.add(array[j]);
            }
            Collections.sort(list);
            answer[i] = list.get(k-1);
        }
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.kt
import java.util.*
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = mutableListOf<Int>()
        var i = 0 
        commands.forEach {
            val slicedList = array.slice(it[0]-1 until it[1])
            val sortedList = slicedList.sorted()
            answer.add(sortedList[it[2]-1])
            i +=1
            
        }
        return answer.toIntArray()
    }
}